test/e2e/flowexporter_test.go

```diff
@@ -55,65 +101,56 @@ func TestFlowExporter(t *testing.T) {
 	}
 
-	// Add NetworkPolicy between two iperf Pods.
-	np1, err := data.createNetworkPolicy("test-networkpolicy-ingress", &networkingv1.NetworkPolicySpec{
-		PodSelector: metav1.LabelSelector{},
-		PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
-		Ingress: []networkingv1.NetworkPolicyIngressRule{{
-			From: []networkingv1.NetworkPolicyPeer{{
-				PodSelector: &metav1.LabelSelector{
-					MatchLabels: map[string]string{
-						"antrea-e2e": "perftest-a",
-					},
-				}},
-			},
-		}},
-	})
+	svcC, err := data.createService("perftest-c", iperfPort, iperfPort, map[string]string{"antrea-e2e": "perftest-c"}, false, v1.ServiceTypeClusterIP)
 	if err != nil {
-		t.Fatalf("Error when creating network policy: %v", err)
+		t.Fatalf("Error when creating perftest service: %v", err)
 	}
-	defer func() {
-		if err = data.deleteNetworkpolicy(np1); err != nil {
-			t.Fatalf("Error when deleting network policy: %v", err)
-		}
-	}()
-	np2, err := data.createNetworkPolicy("test-networkpolicy-egress", &networkingv1.NetworkPolicySpec{
-		PodSelector: metav1.LabelSelector{},
-		PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeEgress},
-		Egress: []networkingv1.NetworkPolicyEgressRule{{
-			To: []networkingv1.NetworkPolicyPeer{{
-				PodSelector: &metav1.LabelSelector{
-					MatchLabels: map[string]string{
-						"antrea-e2e": "perftest-b",
-					},
-				}},
-			},
-		}},
-	})
-	if err != nil {
-		t.Fatalf("Error when creating network policy: %v", err)
+
+	if err := data.createPodOnNode("perftest-c", workerNodeName(1), perftoolImage, nil, nil, nil, []v1.ContainerPort{{Protocol: v1.ProtocolTCP, ContainerPort: iperfPort}}, false, nil); err != nil {
+		t.Fatalf("Error when creating the perftest server Pod: %v", err)
 	}
-	defer func() {
-		if err = data.deleteNetworkpolicy(np2); err != nil {
-			t.Fatalf("Error when deleting network policy: %v", err)
-		}
-	}()
-	// Wait for network policies to be realized.
-	if err := WaitNetworkPolicyRealize(1, data); err != nil {
-		t.Fatalf("Error when waiting for network policy to be realized: %v", err)
+	podCIP, err := data.podWaitForIPs(defaultTimeout, "perftest-c", testNamespace)
+	if err != nil {
+		t.Fatalf("Error when getting the perftest server Pod's IP: %v", err)
 	}
-	t.Log("Network policies are realized.")
-	podAIPStr := podAIP.ipv4.String()
-	podBIPStr := podBIP.ipv4.String()
-	checkRecordsWithPodIPs(t, data, podAIPStr, podBIPStr, false)
+
+	// IntraNodeFlows tests the case, where Pods are deployed on same Node and their flow information is exported as IPFIX flow records.
+	t.Run("IntraNodeFlows", func(t *testing.T) {
+		np1, np2 := deployNetworkPolicy(t, data)
+		defer func() {
+			if err = data.deleteNetworkpolicy(np1); err != nil {
+				t.Fatalf("Error when deleting network policy: %v", err)
+			}
+			if err = data.deleteNetworkpolicy(np2); err != nil {
+				t.Fatalf("Error when deleting network policy: %v", err)
```

Comment:
we do not use `Fatalf` for cleanup as if one operation fails, the rest of the cleanup will be skipped

you can use `Errorf`